acra/guides/advanced-integrations/client-side-integration-with-acra-connector.md

```diff
@@ -0,0 +1,224 @@
+
+#### Writes
+
+When you need to store some sensitive data:
+
+- before doing `INSERT`/`UPDATE` on data, you should use AcraWriter libraries
+  specific to your language, encrypting sensitive data into AcraStruct with AcraServer's
+  public key with suffix `_storage` (`some_client_id_storage.pub`) that is different from the keypair used by AcraConnector for communicating with AcraServer.
+- then you can proceed with doing `INSERT`/`UPDATE`, either through AcraConnector or directly.
+
+#### Reads
+
+Read requests are the same as the regular reads, but their remote address is AcraConnector instead of your database listener. A typical PostgreSQL connection address would look like this: `postgresql://user:password@127.0.0.1:9494/db_name`.
+
+> Note for PostgreSQL: In the current layout, you aren't required to use SSL when trying to connect to the database. Transport protection for sensitive (decrypted) data is provided between AcraServer and AcraConnector via [Secure Session]({{<ref "themis/crypto-theory/cryptosystems/secure-session.md" >}}). However, you can setup using SSL connection, too.
+
+### Client-side with Zones
+
+Description of the client-side with Zones for PostgreSQL.
+
+Zone-based encryption is the best way to cryptographically compartmentalise the data that comes from different sources, following the user's choice. Cryptographically, the Zones are a mechanism for modifying the AcraServer key according to the Zone Id and for mapping Zone's private keys during the decryption. In theory, you could use a new key for every record. However, it would introduce an unwanted overhead on AcraServer's performance.
+
+#### Writing
+
+To write AcraStruct with Zone keys, you need to:
+
+* generate the Zone keys on AcraServer,
+* acquire Zone public key and Zone Id from AcraServer,
+* encrypt data using this key as AcraServer key in a standard AcraStruct process.
+
+There are two ways you can generate the keys:
+
+- First, make sure that you already generated master key and exported it as ACRA_MASTER_KEY environmental variable, or generate it like this:
+```
+export ACRA_MASTER_KEY=$(echo -n "My_Very_Long_Key_Phrase_ge_32_chars" | base64)
+```
+Read more about different types of keys used in Acra on [Key Management]({{< ref "acra/acra-in-depth/cryptography-and-key-management/#-INVALID" >}}) page.
+
+- Then build `acra-addzone` utility in AcraServer from Acra repository:
+
+```
+go build acra-addzone
+./acra-addzone
+``` 
+When you run it, you should get JSON object that looks something like this:
+
+```
+{"id":"DDDDDDDDQHpbUSOgYTzqCktp","public_key":"VUVDMgAAAC3yMBGsAmK/wBXZkL8iBv/C+7hqoQtSZpYoi4fZYMafkJbWe2dL"}
+```
+
+By decoding base64-wrapped public key, you get the binary Zone public key, which can be used for generating AcraStructs.
+
+- AcraConnector locally provides HTTP API to add Zones and return public keys on `127.0.0.1:9191/getNewZone`. You can change the port via `--incoming_connection_api_port` argument.
```

Comment:
> locally provides

Just “provides“ is enough. You can say “provides an HTTP endpoint accessible only from localhost” or something which is more clear.

Reply:
updated